pkg/reconciler/namespace/namespace.go

```diff
@@ -37,28 +38,28 @@ import (
 	"github.com/kcp-dev/kcp/third_party/conditions/util/conditions"
 )
 
-const clusterLabel = "kcp.dev/cluster"
+const (
+	clusterLabel  = "kcp.dev/cluster"
+	scheduleLabel = "kcp.dev/schedule"
+)
 
 var (
-	unscheduledSelector        labels.Selector
-	scheduledToNothingSelector labels.Selector
+	scheduleRequirement    labels.Requirement
+	unscheduledRequirement labels.Requirement
 )
 
 func init() {
-	var err error
-
-	// This matches namespaces without the label
-	unscheduled := "!" + clusterLabel
-	unscheduledSelector, err = labels.Parse(unscheduled)
-	if err != nil {
-		klog.Fatalf("error parsing label selector %q: %v", unscheduled, err)
+	// This matches namespaces that haven't been scheduled yet
```

Comment:
Why does it make sense to make this check more restrictive (label does not exist) than previous (label does not exist or does exist and has a non-empty value)? 

While I can see that your intent is that the controller remove the label, if the label were set to an empty value (e.g. `kcp.dev/cluster=`), the code as proposed would appear neither to ensure removal of the label or ensure queueing of the namespace when cluster readiness status changed.

Reply:
My understanding is that the existing code has to deal with `kcp.dev/cluster=`, because the logic that updates the label does not handle the removal of the label, and simply sets it to `""`. So my rationale was that with properly handling the label deletion, there is no need to deal with `kcp.dev/cluster=` anymore.

Reply:
I'm afraid it's not an option to assume that the label in question will only be managed by this controller. There is always the potential for other actors - including humans - to make changes that don't match this controller's expectations.  Not assuming anything about the input is also a good way to ensure forwards and backwards compatibility.

Reply:
bump

Reply:
True. I've restore the existing logic.